src/index.js

```diff
@@ -27,8 +27,22 @@ function syncEvent(node, eventName, newEventHandler) {
 }
 
 export default function (CustomElement, opts) {
+  // Allows to pass the tagName instead of the CustomElement Class
+  // Avoids creating an instance of the component just to get the tagName.
+  // Makes it possible to use this with Web.Components written in TypeScript.
+  var tagName;
+  if (typeof CustomElement === 'string') {
+    tagName = CustomElement;
+    CustomElement = customElements.get(tagName);
```

Comment:
> it's on window

Yeah, because `window`. It's still probably good to do a `const { customElements } = window;` at the top.

> but do we fill that cross-browser

We don't, no. This was part of the reason that we opted to just use the constructor in the beginning.